src/NuGet.Core/NuGet.Commands/PackCollectorLogger.cs

```diff
@@ -14,14 +14,22 @@ public class PackCollectorLogger : LoggerBase
         private readonly ConcurrentQueue<ILogMessage> _errors;
         private ILogger _innerLogger;
 
+        // Project level warnings properties
         public WarningProperties WarningProperties { get; set; }
 
+        // Package specific warning property
+        public PackCommand.WarningPropertiesCollection PackageReferenceWarningPropertiesCollection { get; set; }
+
         public IEnumerable<ILogMessage> Errors => _errors.ToArray();
 
         public PackCollectorLogger(ILogger innerLogger, WarningProperties warningProperties)
+            : this(innerLogger, warningProperties, null) { }
+
+        public PackCollectorLogger(ILogger innerLogger, WarningProperties warningProperties, PackCommand.WarningPropertiesCollection packageReferenceWarningPropertiesCollection)
```

Comment:
Change it to a using, don't specify `PackCommand.`.

Reply:
I already tried it, but problem is this file is under `namespace NuGet.Commands`, so automatically wrong `WarningPropertiesCollection` from Restore takes into effect (since they reside on same namespace) even though I already defined correct one with using. 
Another solution would be change namespace of this file to  `NuGet.Commands.PackCommand`, but that would trigger breaking change for bunch of public methods/properties in it, not sure if it's worth it.

Reply:
Does WarningPropertiesCollection need a better in that case?

Reply:
I added new `WarningPropertiesCollection` and `PackageSpecificWarningProperties` for PackCommand due this [comment](https://github.com/NuGet/NuGet.Client/pull/4369#discussion_r772549464). I can make this code working `WarningPropertiesCollection`, but I believe `pack` code should follow same pattern as `restore` once.

Reply:
After close examining not much code could be shared, so it better to keep them separate. 
For example: Most methods in Restore>>[WarningPropertiesCollection](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/WarningPropertiesCollection.cs) use `IRestoreLogMessage  message` as parameter, but this Pack>>WarningPropertiesCollection.cs use only `IPackLogMessage message` as parameter.

Another solution would be since I'm adding 2 new files for warnings I can name them different than restore warnings to avoid this situation, without touching any restore files.
1. `NuGet.Commands.PackCommand.PackageSpecificWarningProperties.cs` -> `NuGet.Commands.PackPackageSpecificWarningProperties.cs`
2. `NuGet.Commands.PackCommand.WarningPropertiesCollection` ->
`NuGet.Commands.PackWarningPropertiesCollection.cs`

Reply:
Root of cause this issue is `NuGet.Core\NuGet.Commands\RestoreCommand\Logging\WarningPropertiesCollection.cs` is `NuGet.Commands` under namespace, I wish it was under `NuGet.Commands.Restore` namespace even though it's not really common for both Restore and Pack scenario.